crates/api_models/src/payments.rs

```diff
@@ -343,26 +417,26 @@ pub struct AmountDetails {
     /// The payment amount. Amount for the payment in the lowest denomination of the currency, (i.e) in cents for USD denomination, in yen for JPY denomination etc. E.g., Pass 100 to charge $1.00 and 1 for 1¥ since ¥ is a zero-decimal currency. Read more about [the Decimal and Non-Decimal Currencies](https://github.com/juspay/hyperswitch/wiki/Decimal-and-Non%E2%80%90Decimal-Currencies)
     #[schema(value_type = u64, example = 6540)]
     #[serde(default, deserialize_with = "amount::deserialize")]
-    order_amount: Amount,
+    pub order_amount: Amount,
```

Comment:
We should use `Getters` instead making them as `pub` for all field inside `AmountDetails`

Reply:
We are also constructing this type in router crate though ForeignTryFrom trait implementation . Which is not possible if the fields are private.